vcfconcat.c

```diff
@@ -549,6 +553,114 @@ static void concat(args_t *args)
     }
 }
 
+BGZF *hts_get_bgzfp(htsFile *fp);
+static void naive_concat(args_t *args)
+{
+    // only compressed BCF atm
+
+    BGZF *bgzf_out;
+    if ( strcmp("-",args->output_fname) )
+        bgzf_out = bgzf_open(args->output_fname,"w");
+    else
+        bgzf_out = bgzf_dopen(fileno(stdout), "w");
+
+    int page_size = getpagesize();
+    char *buf = (char*) valloc(page_size);
```

Comment:
Using a page-aligned buffer was important in 1983 when doing page-aligned reads from a file.  Here the reads are not page-aligned and it is no longer 1983.  So either evidence is needed showing some gain from `valloc`, or anything beyond `malloc(some_reasonable_constant)` is premature optimisation.  And 32K is IMHO the absolute minimum reasonable constant.